sycl/source/detail/event_impl.hpp

```diff
@@ -169,6 +170,10 @@ class event_impl {
   /// \return a native handle.
   pi_native_handle getNative() const;
 
+  /// \return a native handle or initializes if it is defaultly constructed
+  /// examplar.
```

Comment:
This reads a little weird. It makes it sound like it only returns the native handle if the event is initialized. Maybe something like the following would make the intention a little clearer?

```suggestion
  /// Gets the native handle associated with this event. If this event is
  /// uninitialized it will first be initialized using the default context for
  /// the specified backend. If the event is initialized by this function, it
  /// is bound to the specified backend.
  ///
  /// \param Backend is the backend to use the default context of if the event
  ///   is uninitialzed.
  /// \return a native handle.
```